src/internal/connector/onedrive/restore.go

```diff
@@ -157,17 +157,17 @@ func RestoreCollection(
 		return metrics, clues.Wrap(err, "creating drive path").WithClues(ctx)
 	}
 
-	if rootIDCache == nil {
-		rootIDCache = map[string]string{}
+	if rootFolderIDCache == nil {
+		rootFolderIDCache = map[string]string{}
```

Comment:
I still feel like this is a code smell. It seems like either we should be passing it in all the time and error out if it's not or we should never pass it in and create it in this function

Reply:
It is a smell, but it gets cleaned up in the next PR.